Tests/Bench/Client/src/Program.cs

```diff
@@ -239,24 +266,62 @@ private static async Task Main()
 
     var resultsAvailable = Stopwatch.GetTimestamp();
 
-    foreach (var resultId in results)
+    var countRes = 0;
+
+    results.AsParallel()
+           .WithDegreeOfParallelism(benchOptions.DegreeOfParallelism)
+           .ForAll(resultId =>
+                   {
+                     for (var i = 0; i < benchOptions.MaxRetries; i++)
+                     {
+                       var localChannel = channelPool.Get();
+                       try
+                       {
+                         var resultRequest = new ResultRequest
+                                             {
+                                               ResultId = resultId,
+                                               Session  = createSessionReply.SessionId,
+                                             };
+
+                         var client = new Submitter.SubmitterClient(localChannel);
+
+                         var result = client.GetResultAsync(resultRequest,
```

Comment:
Even if we are not doing anything with the results afterwards, I think they should be put in a thread safe storage object one they are retrieved, isn't it?

Reply:
I don't want to store them. They can be really large and they will only stop us from running very large tests

Reply:
Ok, I'd then add a comment on the "right way" to store them, in case a future user might use this as en example  of how to recuperate the results.